```
17:05:42 /mdsplus-docker/jenkins/workspace/fc28/src/mdsobjects/cpp/mdsdatastreams.cpp: In static member function 'static void MDSplus::EventStream::send(int, const char*, MDSplus::Data*, MDSplus::Data*)':
17:05:42 /mdsplus-docker/jenkins/workspace/fc28/src/mdsobjects/cpp/mdsdatastreams.cpp:463:12: error: null destination pointer [-Werror=format-overflow=]
17:05:42      sprintf(msgBuf, "%d %s B %d ", shot, name, nTimesSer);
17:05:42      ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
17:05:42 cc1plus: all warnings being treated as errors
```

--- a/mdsobjects/cpp/mdsdatastreams.cpp
+++ b/mdsobjects/cpp/mdsdatastreams.cpp
@@ -456,7 +456,7 @@ EXPORT void EventStream::send(int shot, const char *name, int numSamples, uint64
 
 EXPORT void EventStream::send(int shot, const char *name, Data *timesD, Data *samplesD)
 {
-    int nTimesSer, nSamplesSer;
+    int nTimesSer = 0, nSamplesSer = 0;
     char *timesSer = timesD->serialize(&nTimesSer);
     char *samplesSer = samplesD->serialize(&nSamplesSer);
     char *msgBuf = new char[nTimesSer +  nSamplesSer + 256];

--- a/mdsobjects/cpp/mdsdatastreams.cpp
+++ b/mdsobjects/cpp/mdsdatastreams.cpp
@@ -460,8 +460,10 @@ EXPORT void EventStream::send(int shot, const char *name, Data *timesD, Data *sa
     char *timesSer = timesD->serialize(&nTimesSer);
     char *samplesSer = samplesD->serialize(&nSamplesSer);
     char *msgBuf = new char[nTimesSer +  nSamplesSer + 256];
-    sprintf(msgBuf, "%d %s B %d ", shot, name, nTimesSer);
-    int headerLen = strlen(msgBuf);
+    char headBuf[256];
+    sprintf(headBuf, "%d %s B %d ", shot, name, nTimesSer);
+    int headerLen = strlen(headBuf);
+    memcpy(msgBuf, headBuf, headerLen);
     memcpy(&msgBuf[headerLen], timesSer, nTimesSer);
     memcpy(&msgBuf[headerLen+nTimesSer], samplesSer, nSamplesSer);
     Event::setEventRaw("STREAMING", headerLen+nTimesSer+nSamplesSer, msgBuf);

